client/state/account-recovery/settings/reducer.js

```diff
@@ -1,33 +1,137 @@
 } from 'state/action-types';
 
-const reducer = createReducer( {}, {
-	[ ACCOUNT_RECOVERY_SETTINGS_FETCH ]: ( state ) => ( {
+const isUpdating = createReducer( {}, {
+	[ ACCOUNT_RECOVERY_SETTINGS_UPDATE ]: ( state, { target } ) => ( {
 		...state,
-		isFetching: true,
+		[ target ]: true,
 	} ),
-
-	[ ACCOUNT_RECOVERY_SETTINGS_FETCH_SUCCESS ]: ( state, { email, email_validated, phone, phone_validated } ) => ( {
+	[ ACCOUNT_RECOVERY_SETTINGS_UPDATE_SUCCESS ]: ( state, { target } ) => ( {
 		...state,
-		email,
-		emailValidated: email_validated,
-		phone,
-		phoneValidated: phone_validated,
-		isFetching: false,
+		[ target ]: false,
 	} ),
+	[ ACCOUNT_RECOVERY_SETTINGS_UPDATE_FAILED ]: ( state, { target } ) => ( {
+		...state,
+		[ target ]: false,
+	} ),
+} );
 
-	[ ACCOUNT_RECOVERY_SETTINGS_FETCH_FAILED ]: ( state ) => ( {
+const isDeleting = createReducer( {}, {
+	[ ACCOUNT_RECOVERY_SETTINGS_DELETE ]: ( state, { target } ) => ( {
 		...state,
-		isFetching: false,
+		[ target ]: true,
+	} ),
+	[ ACCOUNT_RECOVERY_SETTINGS_DELETE_SUCCESS ]: ( state, { target } ) => ( {
+		...state,
+		[ target ]: false,
+	} ),
+	[ ACCOUNT_RECOVERY_SETTINGS_DELETE_FAILED ]: ( state, { target } ) => ( {
+		...state,
+		[ target ]: false,
 	} ),
 } );
 
-export default reducer;
+const convertPhoneResponse = ( phoneResponse ) => {
+	if ( ! phoneResponse ) {
+		return null;
+	}
+
+	const {
+		country_code,
+		country_numeric_code,
+		number,
+		number_full,
+	} = phoneResponse;
+
+	return {
+		countryCode: country_code,
+		countryNumericCode: country_numeric_code,
+		number: number,
+		numberFull: number_full,
+	};
+};
+
+const convertEmailResponse = ( emailResponse ) => {
+	return emailResponse ? emailResponse : '';
+};
+
+const data = createReducer( null, {
+	[ ACCOUNT_RECOVERY_SETTINGS_FETCH_SUCCESS ]: ( state, { settings } ) => {
+		const {
+			email,
+			email_validated,
+			phone,
+			phone_validated,
+		} = settings;
+
+		// At the moment, the response from /me/account-recovery could have an object or false for the phone field,
+		// and an string or false for the email field. I personally don't like the mixed value so did the conversion
+		// in the following.
+		return {
+			...state,
+			email: convertEmailResponse( email ),
+			emailValidated: email_validated,
+			phone: convertPhoneResponse( phone ),
+			phoneValidated: phone_validated,
+		};
+	},
+
+	[ ACCOUNT_RECOVERY_SETTINGS_UPDATE_SUCCESS ]: ( state, { target, value } ) => {
+		switch ( target ) {
+			case 'phone':
+				return {
+					...state,
+					phone: convertPhoneResponse( value ),
+				};
+			case 'email':
+				return {
+					...state,
+					email: convertEmailResponse( value ),
+				};
+			default: // do nothing to unknown targets
+				return { ...state };
+		}
+	},
+
+	[ ACCOUNT_RECOVERY_SETTINGS_DELETE_SUCCESS ]: ( state, { target } ) => {
+		switch ( target ) {
+			case 'phone':
+				return {
+					...state,
+					phone: null,
+				};
+			case 'email':
+				return {
+					...state,
```

Comment:
Love the suggestion! Like @omarjackman mentioned, I choose to keep it simple in 75b2fe39319f01ee8256e780b8c2d1ad5868f46f.